src/Featurizers/Components/StandardDeviationUpdater.h

```diff
@@ -0,0 +1,148 @@
+
+        FEATURIZER_MOVE_CONSTRUCTOR_ONLY(STDResult);
+    };
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Types
+    // |
+    // ----------------------------------------------------------------------
+    using InputType                         = T;
+
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    STDUpdater(std::double_t average, std::uint64_t count);
+
+    void update(InputType input);
+
+    STDResult commit(void);
+
+private:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Data
+    // |
+    // ----------------------------------------------------------------------
+    std::double_t  const _average;
+    std::uint64_t  const _count;
+    NormUpdater<std::double_t> _updater;
+};
+
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+// |
+// |  Implementation
+// |
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+
+// ----------------------------------------------------------------------
+// |
+// |  Details::STDUpdater::STDResult
+// |
+// ----------------------------------------------------------------------
+
+template <typename T>
+Details::STDUpdater<T>::STDResult::STDResult(long double standard_deviation) :
+    StandardDeviation(std::move(standard_deviation)) {
+        // input type can only be integer or numeric
+        static_assert(Traits<T>::IsIntOrNumeric::value, "Input type to standard deviation updater has to be integer or numerical types!");
+        if (StandardDeviation < 0) {
+            throw std::invalid_argument("Standard deviation shouldn't be less than 0!");
+        }
+}
+
+// ----------------------------------------------------------------------
+// |
+// |  Details::STDUpdater
+// |
+// ----------------------------------------------------------------------
+
+template <typename T>
+Details::STDUpdater<T>::STDUpdater(std::double_t average, std::uint64_t count) :
+    _average(average),
+    _count(count),
+    _updater(NormUpdater<std::double_t>()) {
+        // input type can only be integer or numeric
+        static_assert(Traits<T>::IsIntOrNumeric::value, "Input type to standard deviation updater has to be integer or numerical types!");
+        if (_count == 0) {
+            throw std::invalid_argument("Count is 0 when constructing STDUpdater to calculate standard deviation!");
+        }
+}
+
+template <typename T>
+void Details::STDUpdater<T>::update(T input) {

```

Comment:
Verify that internal coutner < _count (aka this function should only be called _count times)